### ISSUE TITLE
ChangeNotifier not listened to when overriding ChangeNotifierProvider

### ISSUE THREAD

**Issue by SteveAlexander** [rrousselGit/river_pod#370]
This is a simple app that increments a counter on the screen when the "+1" button is pressed.

It works as expected when the provider is not overridden in the `ProviderScope`. But when it's overridden, the widget no longer updates. I think it's because the provider's new `ChangeNotifier` value is not being listened to.

To reproduce, migrate a default null-safe flutter project, and add `flutter_riverpod: 0.13.1` as a dependency. Then run, and see that tapping the button makes the number go up.

Then, uncomment the override, restart the app, and try again. You will see that the number does not go up.

```dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

class Counter extends ChangeNotifier {
  int _count = 0;
  int get count => _count;

  void increment() {
    _count += 1;
    notifyListeners();
  }
}

final counterProvider = ChangeNotifierProvider((_) => Counter());

void main() {
  runApp(
    ProviderScope(
      overrides: [
        // counterProvider.overrideWithValue(Counter()),
      ],
      child: MyApp(),
    ),
  );
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Material(
        child: Home()
      ),
    );
  }
}

class Home extends ConsumerWidget {
  @override
  Widget build(BuildContext context, ScopedReader watch) {
    final counter = watch(counterProvider);
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('Count = ${counter.count}'),
          TextButton(
            onPressed: () => context.read(counterProvider).increment(),
            child: Icon(Icons.plus_one),
          ),
        ],
      ),
    );
  }
}

```

**Comment by salehahmedZ** [comment]
I faced this issue as well, I had to change my code from overriding provider to direct assignment 


Changed this code:
```dart
final counterProvider = ChangeNotifierProvider((_) => Counter());

void main() {
  runApp(
    ProviderScope(
      overrides: [
        // counterProvider.overrideWithValue(Counter()),
      ],
      child: MyApp(),
    ),
  );
}
```


to this code :
```dart
counterProvider = ChangeNotifierProvider((_) => Counter());

void main() {

// awaited functions

counterProvider = ChangeNotifierProvider((_) => Counter());

  runApp(
    ProviderScope(
      //overrides: [
        // counterProvider.overrideWithValue(Counter()),
     // ],
      child: MyApp(),
    ),
  );
}
```

Not sure if this is good solution but it works fine for me

**Comment by SteveAlexander** [comment]
Thanks — not sure this works well though, as the identity of the provider object will change, i.e. it has become a different `ChangeNotifierProvider` to the one referenced by any code that runs before you reassigned the global.

I expect this will make looking up the provider not work for those cases.

**Comment by salehahmedZ** [comment]
You are right, but I do this only one time before starting the app. 

It's definitely a bug I hope it get fixed.

**Comment by rrousselGit** [comment]
Definitely a bug

But you can use `overrideWIthProvider` instead:

```dart
ProviderScope(
  overrides: [
    counterProvider.overrideWithProvider(
      ChangeNotifierProvider((ref) => Counter()),
    ),
  ],
);
```

**Comment by rrousselGit** [comment]
Should be fixed since 0.14.x
